internal/builders/docker/pkg/builder.go

```diff
@@ -481,25 +481,39 @@ func (c *GitClient) checkoutGitCommit() error {
 func saveToTempFile(verbose bool, readers ...io.Reader) ([]string, error) {
 	var files []string
 	for _, reader := range readers {
-		bytes, err := io.ReadAll(reader)
-		if err != nil {
-			return files, err
-		}
+		var allBytes []byte
 
-		if verbose {
-			if len(bytes) > 0 {
+		for _, reader := range readers {
+			if verbose {
 				fmt.Print("\n\n>>>>>>>>>>>>>> output from command <<<<<<<<<<<<<<\n")
-				fmt.Printf("%s", bytes)
+			}
+
+			scanner := bufio.NewScanner(reader)
+			for scanner.Scan() {
```

Comment:
is there a difference between using scanner instead of the original `fmt.Printf("%s", bytes)`?

Reply:
the scanner incrementally returns individual lines of the reader, we then print each. the original solution read all bytes, and then printed them at once.

Reply:
see https://pkg.go.dev/bufio#Scanner.Split

Reply:
I guess my question is: what's wrong with the current printf? Is it that it may panic of there's too much data to print at once? Something else?

Reply:
There are two downsides that I can think of:

1) (Less of an issue, just a "nice to have") - You can't see any output until the job completes. It's nice to be able to see progress as the task is running to ensure that things are progressing as expected.

2) (The reason we're exploring this) - If something happens to cause the task to hang or take a long time, the job is aborted. Because it's killed, there's never a chance for stdout or stderr to be printed to the output area. This makes it very challenging to understand what happened during the build to cause this. We have a task that's hanging and triggering a timeout. The behavior is only reproducible in this context, but we haven't been able to find a way to get any of the output.

Reply:
Update on this: We've been running our problematic job against this patched builder. The output streams along as the build progresses, which is really nice!

Strangely, the build just succeeds without hanging when we use this patch. I can't quite pin down why that would be the case.